AGS02MA.cpp

```diff
@@ -224,6 +224,24 @@ int AGS02MA::lastError()
 }
 
 
+int AGS02MA::readRegister(uint8_t address, AGS02MA::Register &reg) {
+  if (_readRegister(address))
+  {
+    _error = AGS02MA_OK;
+    reg.data[0] = _buffer[0];
+    reg.data[1] = _buffer[1];
+    reg.data[2] = _buffer[2];
+    reg.data[3] = _buffer[3];
+    reg.crc = _buffer[4];
+    if (_CRC8(_buffer, 5) != 0)
```

Comment:
Same remark here: if CRC fails what is the quality of the individual fields.

Reply:
Indeed, I'll use an early return here like the other PR. And return bool to be consistent.

Reply:
The sketch already refused to display bad CRC data, but now we're also not polluting the passed struct till we've validated the internal buffer.

Example when starting with sensor disconnected:

```
/home/beanow/Arduino/libraries/AGS02MA/examples/AGS02MA_readRegister/AGS02MA_readRegister.ino
AGS02MA_LIB_VERSION: 0.3.0-pre

BEGIN:	0
REG[0x0]	Error:	-12
REG[0x1]	Error:	-12
REG[0x2]	Error:	-12
REG[0x3]	Error:	-12
REG[0x4]	Error:	-12
REG[0x5]	Error:	-12
REG[0x6]	Error:	-12
REG[0x7]	Error:	-12
REG[0x8]	7	118	0	6	C: 105	
REG[0x11]	21	2	3	118	C: 133		Version:	118
REG[0x20]	0	0	135	7	C: 205	
REG[0x21]	26	229	26	229	C: 113		I2C address:	0x1A

REG[0x0]	0	0	7	220	C: 0		Sensor data:	0	2012
REG[0x1]	0	125	5	129	C: 249		Calibration:	125	1409
REG[0x2]	0	100	1	247	C: 148		Calibration:	100	503
REG[0x3]	3	232	0	179	C: 171		Calibration:	1000	179
REG[0x4]	39	16	0	63	C: 122		Calibration:	10000	63
REG[0x5]	238	72	35	40	C: 205	
REG[0x6]	26	0	0	18	C: 213	
REG[0x7]	0	141	246	185	C: 74	
REG[0x8]	185	118	0	6	C: 197	
REG[0x11]	21	2	3	118	C: 133		Version:	118
REG[0x20]	0	0	141	185	C: 131	
REG[0x21]	26	229	26	229	C: 113		I2C address:	0x1A
```